ballerina/tests/15-xa-transaction.bal

```diff
@@ -0,0 +1,159 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/sql;
+
+int DBCLIENT2_PORT = 1522;
+
+type XAResultCount record {
+    int COUNTVAL;
+};
+
+@test:Config {
+    groups: ["transaction", "xa-transaction"]
+}
+function testXATransactionSuccess() returns error? {
+    Client dbClient1 = check new (HOST, USER, PASSWORD, DATABASE, PORT, connectionPool = {maxOpenConnections: 1},
+                                    options = {
+                                        useXADatasource: true
+                                    }
+                                   );
+    Client dbClient2 = check new (HOST, USER, PASSWORD, DATABASE, DBCLIENT2_PORT, connectionPool = {maxOpenConnections: 1},
+                                    options = {
+                                        useXADatasource: true
+                                    }
+                                   );
+
+    transaction {
+        _ = check dbClient1->execute(`insert into Customers (customerId, name, creditLimit, country)
+                                values (1, 'Anne', 1000, 'UK')`);
+        _ = check dbClient2->execute(`insert into Salary (id, value ) values (1, 1000)`);
+        check commit;
+    } on fail {
+        test:assertFail(msg = "Transaction failed");
+    }
+
+    int count1 = check getCustomerCount(dbClient1, 1);
+    int count2 = check getSalaryCount(dbClient2, 1);
+    test:assertEquals(count1, 1, "First transaction failed");
+    test:assertEquals(count2, 1, "Second transaction failed");
+
+    check dbClient1.close();
+    check dbClient2.close();
+}
+
+@test:Config {
+    groups: ["transaction", "xa-transaction"]
+}
+function testXATransactionFailureWithDataSource() returns error? {
+    Client dbClient1 = check new (HOST, USER, PASSWORD, DATABASE, PORT, connectionPool = {maxOpenConnections: 1},
+                                    options = {
+                                        useXADatasource: true
+                                    }
+                                   );
+    Client dbClient2 = check new (HOST, USER, PASSWORD, DATABASE, DBCLIENT2_PORT, connectionPool = {maxOpenConnections: 1},
+                                    options = {
+                                        useXADatasource: true
+                                    }
+                                   );
+
+    transaction {
+        // Intentionally fail first statement
+        _ = check dbClient1->execute(`insert into CustomersTrx (customerId, name, creditLimit, country)
+                                values (30, 'Anne', 1000, 'UK')`);
+        _ = check dbClient2->execute(`insert into Salary (id, value ) values (10, 1000)`);
```

Comment:
```suggestion
        _ = check dbClient2->execute(`insert into Salary (id, value) values (10, 1000)`);
```

Reply:
Fixed in [fec0fa2](https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/467/commits/fec0fa2432c2658c55d89da424d2475f20b399ed)